docs/src/tutorials/custom_impurity_density_pn_junction_lit.jl

```diff
@@ -0,0 +1,143 @@
+# It has to take two arguments. An instance of our model and a point.
+# Since the infinite parallel plate capacitor is configured to be calculated on a cartesian grid,
+# the point has to be a `CartesianPoint`.
+# The function returns the impurity density at the respective position.
+# The returned value has to be in units of 1/m``^{3}``.
+
+#=
+!!! note "Sign of the impurity density"
+    The sign of the impurity density determines whether it is p-type or n-type.
+
+    p-type region <-> negative sign: Holes are the majority carriers and are free to move and diffuse into the n-type region. 
+        Electrons are fixed in the lattice. Thus, a negative fixed space charge density is left behind in the depleted p-type region.
+
+    n-type region <-> positive sign: Electrons are the majority carriers and are free to move and diffuse into the p-type region. 
+        Holes are fixed in the lattice. Thus, a positive fixed space charge density is left behind in the depleted n-type region.
+=#
+
+function SolidStateDetectors.get_impurity_density(
+    cdm::PNJunctionImpurityDensity{T}, 
+    pt::CartesianPoint{T}
+)::T where {T}
+    x = pt[1] # In this example, we only need the `x` coordinate of the point.
+    if x > 0
+        -cdm.p_type_density # p-type region -> electrons are fixed -> negative charge 
+    else 
+        cdm.n_type_density  # n-type region -> holes are fixed -> positive charge
+    end
+end
+
+# ## Assign the Density and Calculate the Fields
+
+# Now, we create an instance of our model with reasonable parameters 
+# (for the geometry and bias voltage of the example detector)
+# and assign it to the detector.
+
+pn_junction_impurity_density = PNJunctionImpurityDensity{Float32}(3e16, 1.5e16)
+
+sim.detector = SolidStateDetector(sim.detector, pn_junction_impurity_density);
+
+# We are now ready to calculate the electric potential. 
+# Depletion handling has to be turned on, via the keyword `depletion_handling = true`,
+# in order to detect the regions where the detector is depleted.
+# The other settings for `calculate_electric_potential!` are very specific for this effectively 1D problem
+# to be calculated in 3D.
+# Afterwards, we also calculate the electric field and plot the 
+# electric potential, impurity scale map and the point type map. 
+# The two latter ons show us where the detector is depleted (`impurity scale == 1`).
+
+calculate_electric_potential!(
+    sim, 
+    convergence_limit = 1e-6,
+    max_tick_distance = (0.002, 1, 1) .* u"cm",
+    min_tick_distance = (1e-7, 1, 1) .* u"m",
+    refinement_limits = [0.005, 0.001],
+    depletion_handling = true,
+    verbose = true
+)
+calculate_electric_field!(sim)
+
+plot(
+    plot(sim.electric_potential, y = 0),
+    plot(sim.imp_scale, y = 0),
+    plot(sim.point_types, y = 0),
+    layout = (3, 1),
+    aspect_ratio = :none,
+    size = (800, 800)
+)
+
+# ## 1D Plots of the P-N Junction
+
+# As this is in principle a 1D simulation, lets better plot the 
```

Comment:
Typo: "lets" -> "let's"
But I would prefer to rephrase this to: simulation, **it is most reasonable to** plot ...